cluster/addons/iso-client/opaque/opaque.go

```diff
@@ -0,0 +1,142 @@
+package opaque
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/transport"
+)
+
+// struct which keeps all information necessary for advertising opaque resources
+type OpaqueIntegerResourceAdvertiser struct {
+	Node   string
+	Name   string
+	Value  string
+	Config *restclient.Config
+}
+
+// body for opaque resource request
+type opaqueIntegerResourceRequest struct {
```

Comment:
nit/naming: want to call this a patchOperation or similar? It doesn't have anything specific to OIR.